src/DotNetWorker/Hosting/WorkerMiddlewareWorkerApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,104 @@
+using Microsoft.Azure.Functions.Worker.Middleware;
+using Microsoft.Azure.Functions.Worker.OutputBindings;
+using Microsoft.Azure.Functions.Worker.Pipeline;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Extensions.Hosting
+{
+    /// <summary>
+    /// Provides extension methods to work with Worker Middleware against a <see cref="IHostBuilder"/>.
+    /// </summary>
+    public static class WorkerMiddlewareWorkerApplicationBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the <see cref="IFunctionsWorkerApplicationBuilder"/> to use the default set of middleware used by the worker, in the following order:
+        /// <list type="number">
+        ///     <item>
+        ///         <description><see cref="OutputBindingsMiddleware"/></description>
+        ///     </item>
+        ///     <item>
+        ///         <description><see cref="FunctionExecutionMiddleware"/></description>
+        ///     </item>
+        /// </list>
+        /// </summary>
+        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/> to configure.</param>
+        /// <returns>The configured <see cref="IFunctionsWorkerApplicationBuilder"/>.</returns>
+        public static IFunctionsWorkerApplicationBuilder UseDefaultWorkerMiddleware(this IFunctionsWorkerApplicationBuilder builder)
+        {
+            return builder.UseOutputBindingsMiddleware()
+                .UseFunctionExecutionMiddleware();
+        }
+
+        /// <summary>
+        /// Configures the <see cref="IFunctionsWorkerApplicationBuilder"/> to use the default <see cref="FunctionExecutionMiddleware"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/> to configure.</param>
+        /// <returns>The configured <see cref="IFunctionsWorkerApplicationBuilder"/>.</returns>
+        public static IFunctionsWorkerApplicationBuilder UseFunctionExecutionMiddleware(this IFunctionsWorkerApplicationBuilder builder)
+        {
+            builder.Services.AddSingleton<FunctionExecutionMiddleware>();
+
+            builder.Use(next =>
+            {
+                return context =>
+                {
+                    var middleware = context.InstanceServices.GetRequiredService<FunctionExecutionMiddleware>();
+
+                    return middleware.Invoke(context);
+                };
+            });
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Configures the <see cref="IFunctionsWorkerApplicationBuilder"/> to use the default <see cref="OutputBindingsMiddleware"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/> to configure.</param>
+        /// <returns>The configured <see cref="IFunctionsWorkerApplicationBuilder"/>.</returns>
+        internal static IFunctionsWorkerApplicationBuilder UseOutputBindingsMiddleware(this IFunctionsWorkerApplicationBuilder builder)
```

Comment:
Could this use the `UseMiddleware` function below? That way, we also end up indirectly testing that :D

Reply:
I knew the question was coming :) 

We could, but didn't really want to have to implement the interface when we already have that plumbing in place for the middleware as is. It's also an example of the pattern you can follow for that style of registration. We can have a test that uses the new model.